lib/Hoymiles/src/parser/StatisticsParser.cpp

```diff
@@ -35,17 +35,28 @@ void StatisticsParser::setByteAssignment(const std::list<byteAssign_t>* byteAssi
 
 void StatisticsParser::clearBuffer()
 {
-    memset(_payloadStatistic, 0, STATISTIC_PACKET_SIZE);
+    memset(_payloadStatistic_off, 0, STATISTIC_PACKET_SIZE);
     _statisticLength = 0;
 }
 
+// If the statistics packet has the full size, we swap and confirm that the data is good (return true)
+bool StatisticsParser::swap()
+{
+    if (_statisticLength == STATISTIC_PACKET_SIZE) {
```

Comment:
You cannot just compare to STATISTIC_PACKET_SIZE  as each inverter sends other amount of bytes. If the program flow reaches the handleResponse method, for sure, all packages are received. At least if the `handleResponse` method returns true. But it might be that the webserver callback occurs between the copying the fragments using the `appendFragment` method.

Reply:
I checked regarding the webserver callback, first I implemented the backbuffer but the issue was still there, so it's not a timing issue.

Reg. Packet size, I guess we'll need to use the size from the inverter then.  Is the packet size for each inverter available somewhere?